### PR TITLE
Add missing PaymentInstrument operations

### PR DESCRIPTION
**Pull request by ashkarpetin**

### CODE REVIEW

File: openapi/paths/payment-instruments.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        schema:
+          oneOf:
+            - $ref: ../components/schemas/PaymentInstruments/PaymentInstrumentCreateToken.yaml
+            - $ref: ../components/schemas/PaymentCards/PaymentCardCreatePlain.yaml
+            - $ref: ../components/schemas/BankAccounts/BankAccountCreatePlain.yaml
+            - $ref: ../components/schemas/PayPalAccount.yaml
+    description: PaymentInstrument resource
+    required: true
+  responses:
+    '200':
```
Comment (slavcodev) [inline openapi/paths/payment-instruments.yaml:58]: I believe this is incorrect status for "update", it should be `303.

File: openapi/paths/payment-instruments.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    - Payment Instruments
+  summary: Create a Payment Instrument
+  operationId: PostPaymentInstrument
+  description: |
+    Create a Payment Instrument
+  requestBody:
+    content:
+      application/json:
+        schema:
+          oneOf:
```
Comment (slavcodev) [inline openapi/paths/payment-instruments.yaml:50]: Please create new definiion for this, it's hard to maintain such complex schemas in paths

File: openapi/paths/payment-instruments.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            items:
+              $ref: ../components/schemas/PaymentInstruments/PaymentInstrument.yaml
+    '401':
+      $ref: ../components/responses/Unauthorized.yaml
+    '403':
+      $ref: ../components/responses/Forbidden.yaml
+post:
+  tags:
+    - Payment Instruments
+  summary: Create a Payment Instrument
```
Comment (slavcodev) [inline openapi/paths/payment-instruments.yaml:42]: It seems the desciption can be more accurate and mention the update scenario.

File: openapi/combined.yaml [hunk 4]
```diff
@@ -362,8 +362,12 @@ paths:
     $ref: ./paths/password-tokens.yaml
   '/password-tokens/{id}':
     $ref: './paths/password-tokens@{id}.yaml'
+  '/payment-instruments':
+    $ref: './paths/payment-instruments.yaml'
   '/payment-instruments/{id}':
     $ref: './paths/payment-instruments@{id}.yaml'
+  '/payment-instruments/{id}/deactivation':
```
Comment (slavcodev) [inline openapi/combined.yaml:369]: I woudl recommend remove this RPC-style API.
The `PATCH payment-instruments/{id}` with `"status": "deactivated"` works just fine.

File: openapi/components/schemas/BankAccounts/BankAccountUpdatePlain.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+type: object
```
Comment (slavcodev) [inline openapi/components/schemas/BankAccounts/BankAccountUpdatePlain.yaml:1]: What about "browserData" is this field "patchable"?

Comment (ashkarpetin) [reply openapi/components/schemas/BankAccounts/BankAccountUpdatePlain.yaml:1]: I don't know

Comment (ashkarpetin) [reply openapi/components/schemas/BankAccounts/BankAccountUpdatePlain.yaml:1]: There is no need for this, at least for now

File: openapi/paths/payment-instruments@{id}.yaml [hunk 6]
```diff
@@ -32,25 +33,12 @@ patch:
...
-            token:
-              description: Payment instrument token ID
-              type: string
-            billingAddress:
-              description: The billing address (if supplied – overrides billing address from token).
-              allOf:
-                - $ref: ../components/schemas/Contact/ContactObject.yaml
-            customFields:
-              $ref: ../components/schemas/ResourceCustomFields.yaml
+          $ref: ../components/schemas/PaymentInstruments/UpdatePaymentInstrument.yaml
```
Comment (slavcodev) [inline openapi/paths/payment-instruments@{id}.yaml:41]:
```suggestion
          $ref: ../components/schemas/PaymentInstruments/PaymentInstrumentPatch.yaml
```

Please consider this name

- We are talking about resources and not actions in API
- `Update` may confuse that this schema is also used in PUT

**Review by slavcodev — DISMISSED** [review]

**Review by iamdevice — DISMISSED** [review]

**Review by iamdevice — DISMISSED** [review]

**Review by slavcodev — DISMISSED** [review]

File: openapi/components/schemas/PaymentCards/PaymentCard.yaml [hunk 7]
```diff
@@ -1,126 +1,38 @@
...
-    minItems: 1
-    items:
-      anyOf:
-        - $ref: ../Embeds/AuthTransactionEmbed.yaml
-        - $ref: ../Embeds/CustomerEmbed.yaml
+allOf:
+  - $ref: ../Common/CommonPaymentCard.yaml
+  - properties:
+      stickyGatewayAccountId:
+        description: Default Gateway Account ID used for transactions.
```
Comment (vykhrystiuk) [inline openapi/components/schemas/PaymentCards/PaymentCard.yaml:5]: Should it be an upper case? 
```suggestion
        description: Default gateway account ID used for transactions.
```

File: openapi/components/schemas/PaymentCards/PaymentCardUpdatePlain.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    type: integer
+  expYear:
+    description: Card's expiration year.
+    type: integer
+  billingAddress:
+    description: The billing address.
+    allOf:
+      - $ref: ../Contact/ContactObject.yaml
+  stickyGatewayAccountId:
+    description: Sticky Gateway Account ID.
```
Comment (vykhrystiuk) [inline openapi/components/schemas/PaymentCards/PaymentCardUpdatePlain.yaml:17]:
```suggestion
    description: Sticky gateway account ID.
```

File: openapi/components/schemas/PaymentInstruments/PaymentInstrumentCreateToken.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+type: object
+required:
+  - customerId
+  - token
+properties:
+  customerId:
+    description: The Customer's ID.
```
Comment (vykhrystiuk) [inline openapi/components/schemas/PaymentInstruments/PaymentInstrumentCreateToken.yaml:7]:
```suggestion
    description: The customer's ID.
```

File: openapi/paths/payment-instruments.yaml [hunk 10]
```diff
@@ -0,0 +1,83 @@
+parameters:
+  - $ref: ../components/parameters/organizationId.yaml
+get:
+  tags:
+    - Payment Instruments
+  summary: Retrieve a list of Payment Instruments
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:6]:
```suggestion
  summary: Retrieve a list of payment instruments
```

File: openapi/paths/payment-instruments.yaml [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  operationId: GetPaymentInstrumentCollection
+  description: |
+    Retrieve a list of Payment Instruments
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:9]:
```suggestion
    Retrieve a list of payment instruments
```

File: openapi/paths/payment-instruments.yaml [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  parameters:
+    - $ref: ../components/parameters/collectionLimit.yaml
+    - $ref: ../components/parameters/collectionOffset.yaml
+  responses:
+    '200':
+      description: A list of Payment Instruments was retrieved successfully
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:15]:
```suggestion
      description: A list of payment instruments was retrieved successfully
```

File: openapi/paths/payment-instruments.yaml [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+      $ref: ../components/responses/Unauthorized.yaml
+    '403':
+      $ref: ../components/responses/Forbidden.yaml
+post:
+  tags:
+    - Payment Instruments
+  summary: Create a Payment Instrument
+  operationId: PostPaymentInstrument
+  description: |
+    Create a Payment Instrument.
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:45]:
```suggestion
    Create a payment instrument.
```

File: openapi/paths/payment-instruments@{id}@deactivation.yaml [hunk 14]
```diff
@@ -0,0 +1,32 @@
+parameters:
+  - $ref: ../components/parameters/resourceId.yaml
+  - $ref: ../components/parameters/organizationId.yaml
+post:
+  tags:
+    - Payment Instruments
+  summary: Deactivate a Payment Instrument
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments@{id}@deactivation.yaml:7]:
```suggestion
  summary: Deactivate a payment instrument
```

File: openapi/paths/payment-instruments@{id}@deactivation.yaml [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+  operationId: PostPaymentInstrumentDeactivation
+  description: |
+    Deactivate a Payment Instrument
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments@{id}@deactivation.yaml:10]:
```suggestion
    Deactivate a payment instrument
```

File: openapi/paths/payment-instruments@{id}@deactivation.yaml [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+  responses:
+    '201':
+      description: Payment Instrument was deactivated
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments@{id}@deactivation.yaml:13]:
```suggestion
      description: Payment instrument was deactivated
```

File: openapi/paths/payment-instruments.yaml [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+        Rate-Limit-Remaining:
+          $ref: ../components/headers/Rate-Limit-Remaining.yaml
+        Rate-Limit-Reset:
+          $ref: ../components/headers/Rate-Limit-Reset.yaml
+      content:
+        application/json:
+          schema:
+            $ref: ../components/schemas/PaymentInstruments/PaymentInstrument.yaml
+    '303':
+      description: PaymentInstrument was updated
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:64]:
```suggestion
      description: Payment instrument was updated.
```

?

File: openapi/paths/payment-instruments.yaml [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+  summary: Create a Payment Instrument
+  operationId: PostPaymentInstrument
+  description: |
+    Create a Payment Instrument.
+    If such payment card or bank account payment instrument already exists then updates it instead.
+  requestBody:
+    $ref: ../components/requestBodies/PostPaymentInstrument.yaml
+  responses:
+    '201':
+      description: PaymentInstrument was created
```
Comment (vykhrystiuk) [inline openapi/paths/payment-instruments.yaml:51]:
```suggestion
      description: Payment instrument was created.
```

File: openapi/paths/payment-instruments.yaml (hunk 13 above)
Comment (vykhrystiuk) [reply openapi/paths/payment-instruments.yaml:45]: I don't know if we need a period here. We need it only if it's a sentence.

File: openapi/components/schemas/PaymentInstruments/PaymentInstrumentCreateToken.yaml (hunk 9 above)
Comment (vykhrystiuk) [reply openapi/components/schemas/PaymentInstruments/PaymentInstrumentCreateToken.yaml:7]: I'm not sure we need a period here though

**Review by slavcodev — APPROVED** [review]

**Review by vykhrystiuk — APPROVED** [review]

**Review by iamdevice — APPROVED** [review]

**Pull request merged**